sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,743 @@
+}
+
+/**
+ * Get `num` distinct random numbers in the range [0, `max`] from
+ * RV_CORE_IBEX_RND_DATA.
+ *
+ * @param ibex The Ibex DIF object.
+ * @param num The number of random numbers to get.
+ * @param[out] rnd_buf Pointer to the buffer to write the random numbers to.
+ * @param max The maximum random value returned.
+ */
+static void get_rand_words(dif_rv_core_ibex_t *ibex, int num, uint32_t *rnd_buf,
+                           uint32_t max) {
+  uint32_t rnd_word;
+  for (int i = 0; i < num; ++i) {
+    bool found = false;
+    while (found == false) {
+      // Get a new random number.
+      CHECK_DIF_OK(dif_rv_core_ibex_read_rnd_data(ibex, &rnd_word));
+      rnd_word = rnd_word % max;
+      // Check if the number is unique.
+      found = true;
+      for (int j = 0; j < i; ++j) {
+        if (rnd_buf[j] == rnd_word) {
+          // Start over.
+          found = false;
+          break;
+        }
+      }
+    }
+    // Add the number to the buffer.
+    rnd_buf[i] = rnd_word;
+  }
+}
+
+enum {
+  // Non-volatile counter for the test steps.
+  kCounterTestSteps = 0,
+};
+
+// This function will run after every device reset
+bool test_main(void) {
+  init_peripherals();
+
+  // A generic loop variable
+  int ii;
+  // To keep the test results
+  bool is_cause;
+
+  // Get the random wait times
+  uint32_t rnd_wait_times[ARRAYSIZE(kPeripherals)];
+  uint32_t max;
+  max = 0x1 << 10;
+  get_rand_words(&ibex, ARRAYSIZE(kPeripherals), rnd_wait_times, max);
+
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  // Each test phase consists of ARRAYSIZE(kPeripherals) steps
+  // At every step, a specific peripheral is tested
+  size_t test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+  size_t test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+  size_t peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+
+  // At the beginning of the simulation (Verilator, VCS),
+  // the content of the flash might be all-zeros, and thus,
+  // the NVM counter's inital value might be 256.
+  // In that case, the NVM counter should be initialized to zero by filling
+  // its flash region with non-zero values.
+  // On FPGA, this code will be skipped
+  if (test_step_cnt == 256) {
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestSteps);
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+    peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+  }
+
+  /* TEST PHASE #0: NEGATIVE TESTING
```

Comment:
I wanted to keep this just in case. I can remove this part if we want to stick with the official test description.

Reply:
I think it is ok to keep if the simulation time stays within reasonable bounds. do you have an estimate for that?

Reply:
It takes around 100 seconds on the FPGA. I haven't tried it on DV. I will check it and update the comment.